core/src/main/java/org/infinispan/AdvancedCache.java

```diff
@@ -551,4 +551,17 @@
     * @return the entry set containing all of the CacheEntries
     */
    CacheSet<CacheEntry<K, V>> cacheEntrySet();
+
+   /**
+    * Attempts to remove the entry if it is expired.  Due to expired entries not being consistent across nodes, this
+    * will still attempt to remove the value if it is not present.  Note that this will raise an expired event
+    * even if the entry is not present.  Normally this method should never be invoked except by the
+    * {@link ExpirationManager}.
```

Comment:
If this method is only to be executed by ExpirationManager, why expose it in AdvancedCache? Users have easy access to AdvancedCache. Couldn't ExpirationManager call this directly without the need of AdvancedCache? IOW, if it has all the components available, it could just call this code below without the need to add another method to a public interface?

```
+      InvocationContext ctx = getInvocationContextWithImplicitTransaction(false, null, 1);
+      RemoveExpiredCommand command = commandsFactory.buildRemoveExpiredCommand(key, value, lifespan);
+      ctx.setLockOwner(command.getKeyLockOwner());
+      // Send an expired remove command to everyone
+      executeCommandAndCommitIfNeeded(ctx, command);
```